src/runtime_src/core/edge/drm/zocl/edge/zocl_aie.c

```diff
@@ -244,8 +244,47 @@ zocl_cleanup_aie(struct drm_zocl_dev *zdev)
 	return 0;
 }
 
+int 
+zocl_read_aieresbin(struct drm_zocl_dev *zdev, struct axlf* axlf, char __user *xclbin, enum axlf_section_kind kind)
+{
+	struct axlf_section_header *header = NULL;
+	header = xrt_xclbin_get_section_hdr_next(axlf, AIE_RESOURCES_BIN, header);
+	while (header) {
+		void* aie_res = NULL;	
+		int err = 0;
+
+		aie_res = vmalloc(header->m_sectionSize);
+		err = copy_from_user(aie_res, &xclbin[header->m_sectionOffset], header->m_sectionSize);
+		if (err) {
+			vfree(aie_res);
+			return 0;
+		}
+		
+		struct aie_resources_bin *aie_p = (struct aie_resources_bin *)aie_res;
+       		char *data_portion = (char *)aie_p + aie_p->m_image_offset; 	
+		size_t data_size = aie_p->m_image_size;
+        	void *res = NULL;
+        
+        	res = kmalloc(data_size, GFP_KERNEL);
```

Comment:
cant u directly pass data_portion?

Reply:
The size of the data being sent to the AIE driver is not explicitly defined by passing data_portion alone. Basically the aie driver function will not know the size of the data.